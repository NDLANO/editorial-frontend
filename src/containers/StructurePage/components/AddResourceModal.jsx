import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectT } from 'ndla-i18n';
import TaxonomyLightbox from './TaxonomyLightbox';
import { AsyncDropdown } from '../../../components/Dropdown';
import { groupSearch } from '../../../modules/search/searchApi';
import {
  createTopicResource,
  fetchResource,
  fetchResourceResourceType,
} from '../../../modules/taxonomy';
import { getResourceIdFromPath } from '../../../util/routeHelpers';

import { getArticle } from '../../../modules/article/articleApi';
import ArticlePreview from './ArticlePreview';

class AddResourceModal extends Component {
  constructor() {
    super();
    this.state = {
      selected: {},
      article: {},
      pastedUrl: '',
    };
    this.onSelect = this.onSelect.bind(this);
    this.addSelected = this.addSelected.bind(this);
    this.onPaste = this.onPaste.bind(this);
    this.articleToState = this.articleToState.bind(this);
  }

  onSelect(selected) {
    if (selected) {
      const articleId = selected.url.split('/').pop();
      this.articleToState(articleId);
      this.setState({ selected });
    } else {
      this.setState({ selected: {}, article: {} });
    }
  }

  async onPaste(e) {
    const val = e.target.value;
    const resourceId = getResourceIdFromPath(val);

    if (resourceId) {
      try {
        const [resource, resourceType] = await Promise.all([
          fetchResource(resourceId),
          fetchResourceResourceType(resourceId),
        ]);
        this.articleToState(resource.contentUri.split(':').pop());
        const type = resourceType.length > 0 && resourceType[0].id;
        const error =
          type === this.props.type
            ? ''
            : `${this.props.t('taxonomy.wrongType')} ${type}`;
        this.setState({ selected: { id: val }, pastedUrl: val, error });
      } catch (error) {
        console.log(error);
      }
    } else if (!val) {
      this.setState({ error: '', pastedUrl: val });
    } else {
      this.setState({ error: 'Invalid url', pastedUrl: val });
    }
  }

  async articleToState(articleId) {
    const {
      id,
      metaDescription = {},
      title = {},
      metaImage = {},
    } = await getArticle(articleId);
    this.setState({
      article: {
        id,
        metaDescription: metaDescription.metaDescription,
        title: title.title,
        imageUrl: metaImage.url,
      },
    });
  }

  async addSelected() {
    const { topicId, refreshResources, onClose } = this.props;
    const { selected } = this.state;
    if (selected.id) {
      try {
        this.setState({ loading: true });
        await createTopicResource({
          resourceId: getResourceIdFromPath(selected.id),
          topicid: `urn:${topicId}`,
        });
        refreshResources();
        this.setState({ loading: false });
        onClose();
      } catch (e) {
        console.log(e);
        this.setState({ loading: false, error: e.message });
      }
    }
  }

  render() {
    const { onClose, type, t } = this.props;
    const { selected, article, loading, pastedUrl, error } = this.state;

    return (
      <TaxonomyLightbox
        title={t('taxonomy.searchResource')}
        onSelect={this.addSelected}
        loading={loading}
        onClose={onClose}>
        <input
          type="text"
          data-testid="addResourceUrlInput"
          value={pastedUrl}
          onChange={this.onPaste}
          placeholder={t('taxonomy.urlPlaceholder')}
        />
        {error && <span className="c-errorMessage">{error}</span>}

        {!pastedUrl && (
          <React.Fragment>
            <span
              style={{
                color: 'white',
                alignSelf: 'center',
              }}>
              {t('taxonomy.or')}
            </span>
            <AsyncDropdown
              valueField="id"
              name="resourceSearch"
              textField="title"
              placeholder={t('form.content.relatedArticle.placeholder')}
              label="label"
              apiAction={async inp => {
                const res = await groupSearch(inp, type);
                const result = res.length > 0 ? res.pop().results : [];
                return result.map(curr => ({
                  ...curr,
                  id: curr.paths.pop(),
                  title: curr.title ? curr.title.title : '',
                }));
              }}
              messages={{
                emptyFilter: '',
                emptyList: t('taxonomy.noResources'),
              }}
              onChange={this.onSelect}
              alwaysOpen
            />
          </React.Fragment>
        )}
        {selected.id && article.id && <ArticlePreview article={article} />}
      </TaxonomyLightbox>
    );
  }
}

AddResourceModal.propTypes = {
  onClose: PropTypes.func,
  type: PropTypes.string,
};

export default injectT(AddResourceModal);
