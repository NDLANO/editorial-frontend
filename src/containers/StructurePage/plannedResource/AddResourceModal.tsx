/**
 * Copyright (c) 2023-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import { ReactNode } from "react";
import styled from "@emotion/styled";
import { spacing } from "@ndla/core";

const StyledContent = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: ${spacing.small};
  > * {
    width: 100%;
  }
  & form {
    background-color: white;
  }
`;

interface Props {
  children: ReactNode;
}

const AddResourceModal = ({ children }: Props) => {
  return <StyledContent>{children}</StyledContent>;
};

export default AddResourceModal;
