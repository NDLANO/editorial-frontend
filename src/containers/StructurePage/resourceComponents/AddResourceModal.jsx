import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { injectT } from '@ndla/i18n';
import { Input } from '@ndla/forms';
import styled from '@emotion/styled';
import handleError from '../../../util/handleError';
import TaxonomyLightbox from '../../../components/Taxonomy/TaxonomyLightbox';
import { AsyncDropdown } from '../../../components/Dropdown';
import { groupSearch } from '../../../modules/search/searchApi';
import {
  createTopicResource,
  fetchResource,
  fetchResourceResourceType,
  updateTopic,
} from '../../../modules/taxonomy';
import { getResourceIdFromPath } from '../../../util/routeHelpers';

import { getArticle } from '../../../modules/article/articleApi';
import { learningpathSearch } from '../../../modules/learningpath/learningpathApi';
import ArticlePreview from '../../../components/ArticlePreview';

const StyledOrDivider = styled.div`
  display: flex;
  justify-content: center;
  margin: 20px 0;
`;

const StyledContent = styled.div`
  width: 100%;

  > * {
    width: 100%;
  }

  & form {
    background-color: white;
  }
`;
class AddResourceModal extends Component {
  constructor() {
    super();
    this.state = {
      selected: {},
      article: {},
      pastedUrl: '',
    };
  }

  onSelect = selected => {
    console.log(selected);
    if (selected) {
      if (selected.url && !selected.url.includes('learningpaths')) {
        const articleId = selected.url.split('/').pop();
        this.articleToState(articleId);
      }
      this.setState({ selected });
    } else {
      this.setState({ selected: {}, article: {} });
    }
  };

  onPaste = async e => {
    const val = e.target.value;
    const { type, t } = this.props;
    const resourceId = getResourceIdFromPath(val);

    if (resourceId) {
      try {
        const [resource, resourceType] = await Promise.all([
          fetchResource(resourceId),
          fetchResourceResourceType(resourceId),
        ]);
        this.articleToState(resource.contentUri.split(':').pop());
        const pastedType = resourceType.length > 0 && resourceType[0].id;
        const error =
          pastedType === type ? '' : `${t('taxonomy.wrongType')} ${pastedType}`;
        this.setState({ selected: { id: val }, pastedUrl: val, error });
      } catch (error) {
        handleError(error);
        this.setState({ error: error.message });
      }
    } else if (!val) {
      this.setState({ error: '', pastedUrl: val });
    } else {
      this.setState({ error: t('errorMessage.invalidUrl'), pastedUrl: val });
    }
  };

  onInputSearch = async input => {
    try {
      const result =
        this.props.type === 'urn:resourcetype:learningPath'
          ? await this.searchLearningpath(input)
          : await this.groupSearch(input);
      return result.map(current => ({
        ...current,
        title: current.title ? current.title.title : '',
      }));
    } catch (e) {
      handleError(e);
      this.setState({ error: e.message });
      return [];
    }
  };

  searchLearningpath = async input => {
    const query = {
      query: input,
      pageSize: 10,
      language: 'nb',
      fallback: true,
    };
    const res = await learningpathSearch(query);
    return res.results ? res.results : [];
  };

  groupSearch = async input => {
    const res = await groupSearch(input, this.props.type);
    return res.length > 0 ? res.pop().results : [];
  };

  articleToState = async articleId => {
    const {
      id,
      metaDescription = {},
      title = {},
      metaImage = {},
    } = await getArticle(articleId);
    this.setState({
      article: {
        id,
        metaDescription: metaDescription.metaDescription,
        title: title.title,
        imageUrl: metaImage.url,
      },
    });
  };

  addSelected = async () => {
    const { topicId, refreshResources, onClose } = this.props;
    const { selected } = this.state;
    if (selected.id) {
      try {
        this.setState({ loading: true });
        console.log(selected);
        await updateTopic({
          id: selected.id,
          name: selected.title,
          contentUri: `urn:learningpath:${selected.id}`,
        });
        await createTopicResource({
          resourceId: getResourceIdFromPath(selected.paths[0]),
          topicid: topicId,
        });
        refreshResources();
        this.setState({ loading: false });

        onClose();
      } catch (e) {
        handleError(e);
        this.setState({ loading: false, error: e.message });
      }
    }
  };

  render() {
    const { onClose, t, allowPaste } = this.props;
    const { selected, article, loading, pastedUrl, error } = this.state;
    return (
      <TaxonomyLightbox
        title={t('taxonomy.searchResource')}
        onSelect={this.addSelected}
        loading={loading}
        onClose={onClose}>
        <StyledContent>
          {allowPaste && (
            <Input
              type="text"
              data-testid="addResourceUrlInput"
              value={pastedUrl}
              onChange={this.onPaste}
              placeholder={t('taxonomy.urlPlaceholder')}
            />
          )}
          {error && <span className="c-errorMessage">{error}</span>}

          {!pastedUrl && (
            <React.Fragment>
              {allowPaste && (
                <StyledOrDivider>{t('taxonomy.or')}</StyledOrDivider>
              )}
              <AsyncDropdown
                valueField="id"
                name="resourceSearch"
                textField="title"
                placeholder={t('form.content.relatedArticle.placeholder')}
                label="label"
                apiAction={this.onInputSearch}
                onChange={this.onSelect}
                startOpen
              />
            </React.Fragment>
          )}
          {selected.id && article.id && <ArticlePreview article={article} />}
        </StyledContent>
      </TaxonomyLightbox>
    );
  }
}

AddResourceModal.propTypes = {
  onClose: PropTypes.func,
  type: PropTypes.string,
  allowPaste: PropTypes.bool,
  topicId: PropTypes.string.isRequired,
  refreshResources: PropTypes.func.isRequired,
};

export default injectT(AddResourceModal);
