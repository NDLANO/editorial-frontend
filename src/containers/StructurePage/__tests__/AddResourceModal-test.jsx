/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import nock from 'nock';
import { render, fireEvent, cleanup, wait } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AddResourceModal from '../resourceComponents/AddResourceModal';
import IntlWrapper from '../../../util/__tests__/IntlWrapper';
import { resourcesByType, articleMock } from '../../../util/__tests__/taxonomyMocks';

afterEach(cleanup);

const resourceType = 'urn:resourcetype:reviewResource';
const ndlaUrl = 'https://beta.ndla.no/subject:3/topic:1:179373/topic:1:170165/resource:1:168388/';
const resourceMock = {
  id: 'urn:resource:1:168388',
  name: 'Oppgaver til utforskeren',
  contentUri: 'urn:article:24',
  path: '/subject:9/topic:1:179373/topic:1:170165/resource:1:168388',
};
const resourceTypeMock = [
  {
    id: 'urn:resourcetype:tasksAndActivities',
    parentId: null,
    name: 'Oppgaver og aktiviteter',
    connectionId: 'urn:resource-resourcetype:9414c99b-73c4-4222-8b94-58a84aba02cd',
  },
];

beforeEach(() => {
  nock('http://ndla-api')
    .get(`/search-api/v1/search/group/?query=&resource-types=${resourceType}`)
    .reply(200, resourcesByType);
});

const wrapper = props =>
  render(
    <IntlWrapper>
      <MemoryRouter>
        <AddResourceModal
          topicId="topicId2"
          allowPaste
          t={() => 'injected'}
          type={resourceType}
          onClose={() => {}}
          refreshResources={() => {}}
          startOpen
          {...props}
        />
      </MemoryRouter>
    </IntlWrapper>,
  );

test('Can select a resource from the list and it adds it to topic', async () => {
  nock('http://ndla-api')
    .get(`/article-api/v2/articles/356?language=nb&fallback=true`)
    .reply(200, articleMock);
  nock('http://ndla-api')
    .post(
      '/taxonomy2/v1/topic-resources',
      JSON.stringify({
        resourceId: 'urn:resource:1:175733',
        topicid: 'urn:topicId2',
      }),
    )
    .reply(201);
  const { container, getByText, getByTestId } = wrapper();
  await wait(() => getByText(resourcesByType[0].results[0].title.title));

  expect(container.firstChild).toMatchSnapshot();
  fireEvent.click(getByText(resourcesByType[0].results[0].title.title));
  await wait(() => getByTestId('articlePreview'));

  expect(container.firstChild).toMatchSnapshot();
  fireEvent.click(getByTestId('taxonomyLightboxButton'));
  await wait();
  expect(nock.isDone());
});

test('Can paste a valid url and add it to topic', async () => {
  nock('http://ndla-api')
    .get(`/article-api/v2/articles/24?language=nb&fallback=true`)
    .reply(200, articleMock);
  nock('http://ndla-api')
    .get(`/taxonomy2/v1/resources/urn:resource:1:168388`)
    .reply(200, resourceMock);
  nock('http://ndla-api')
    .get(`/taxonomy2/v1/resources/urn:resource:1:168388/resource-types/`)
    .reply(200, resourceTypeMock);
  nock('http://ndla-api')
    .post(
      '/taxonomy2/v1/topic-resources',
      JSON.stringify({
        resourceId: 'urn:resource:1:168388',
        topicid: 'urn:topicId2',
      }),
    )
    .reply(201);
  const { container, getByTestId } = wrapper();
  const input = getByTestId('addResourceUrlInput');
  fireEvent.change(input, { target: { value: ndlaUrl } });

  await wait(() => getByTestId('articlePreview'));
  expect(container.firstChild).toMatchSnapshot();
  fireEvent.click(getByTestId('taxonomyLightboxButton'));
  await wait();

  expect(nock.isDone());
});
