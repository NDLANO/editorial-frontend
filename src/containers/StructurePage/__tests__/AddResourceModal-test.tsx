/**
 * Copyright (c) 2017-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import nock from 'nock';
import { render, fireEvent, cleanup, act } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';
import AddResourceModal from '../resourceComponents/AddResourceModal';
import IntlWrapper from '../../../util/__tests__/IntlWrapper';
import { resourcesByType, articleMock } from '../../../util/__tests__/taxonomyMocks';
import { taxonomyApi } from '../../../config';

afterEach(cleanup);

const resourceType = 'urn:resourcetype:tasksAndActivities';
const ndlaUrl = 'https://beta.ndla.no/subject:3/topic:1:179373/topic:1:170165/resource:1:168388/';
const resourceMock = {
  id: 'urn:resource:1:168388',
  name: 'Oppgaver til utforskeren',
  contentUri: 'urn:article:24',
  path: '/subject:9/topic:1:179373/topic:1:170165/resource:1:168388',
  resourceTypes: [
    {
      id: 'urn:resourcetype:task',
      parentId: 'urn:resourcetype:tasksAndActivities',
      name: 'Oppgave',
      connectionId: 'urn:resource-resourcetype:9115ed35-3a9b-41e8-9f7a-0717d2ee8146',
    },
    {
      id: 'urn:resourcetype:tasksAndActivities',
      parentId: null,
      name: 'Oppgaver og aktiviteter',
      connectionId: 'urn:resource-resourcetype:0e1ac7ac-fc7d-44db-b217-ebf262ede263',
    },
  ],
};

beforeEach(() => {
  nock('http://ndla-api')
    .get(
      `/search-api/v1/search/group/?fallback=true&language=nb&page=1&query=&resource-types=${encodeURIComponent(
        resourceType,
      )}`,
    )
    .reply(200, resourcesByType);
});

const qc = new QueryClient();

const wrapper = (existingResourceIds: string[] = []) =>
  render(
    <IntlWrapper>
      <MemoryRouter>
        <QueryClientProvider client={qc}>
          <AddResourceModal
            data-testid="addResourceModal"
            nodeId="topicId2"
            allowPaste
            type={resourceType}
            onClose={() => {}}
            existingResourceIds={existingResourceIds}
          />
        </QueryClientProvider>
      </MemoryRouter>
    </IntlWrapper>,
  );

test('Can select a resource from the list and it adds it to topic', async () => {
  nock('http://ndla-api')
    .get(`/article-api/v2/articles/356?language=nb&fallback=true`)
    .reply(200, articleMock);
  nock('http://ndla-api')
    .post(
      `${taxonomyApi}/node-resources`,
      JSON.stringify({
        resourceId: 'urn:resource:1:175733',
        nodeId: 'topicId2',
      }),
    )
    .reply(201, undefined, { Location: 'urn' });
  const { container, getByText, getByTestId, findByText, findByTestId } = wrapper();
  await findByText('Hva kan du om geologiske prosesser?');

  expect(container.firstChild).toMatchSnapshot();
  act(() => {
    fireEvent.click(getByText('Hva kan du om geologiske prosesser?'));
  });
  await findByTestId('articlePreview');

  expect(container.firstChild).toMatchSnapshot();
  act(() => {
    fireEvent.click(getByTestId('taxonomyLightboxButton'));
  });
  await findByText('Lagre');
  expect(nock.isDone());
});

test('Can paste a valid url and add it to topic', async () => {
  nock('http://ndla-api')
    .get(`/article-api/v2/articles/24?language=nb&fallback=true`)
    .reply(200, articleMock);
  nock('http://ndla-api')
    .get(`${taxonomyApi}/resources/urn:resource:1:168388`)
    .reply(200, resourceMock);
  nock('http://ndla-api')
    .post(
      `${taxonomyApi}/node-resources`,
      JSON.stringify({
        resourceId: 'urn:resource:1:168388',
        nodeId: 'topicId2',
      }),
    )
    .reply(201, {}, { Location: 'urn' });
  const {
    container,
    getByTestId,
    findByTestId,
    findByText,
    getByPlaceholderText,
    queryByPlaceholderText,
  } = wrapper();
  await findByText('Hva kan du om geologiske prosesser?');
  const input = getByPlaceholderText('Lim inn lenke fra ndla.no');
  fireEvent.change(input, { target: { value: ndlaUrl } });
  await findByTestId('articlePreview');
  expect(queryByPlaceholderText('Søk på tittel')).not.toBeInTheDocument();
  await findByText(articleMock.title.title);

  expect(container.firstChild).toMatchSnapshot();
  act(() => {
    fireEvent.click(getByTestId('taxonomyLightboxButton'));
  });

  expect((await findByTestId('taxonomyLightboxButton')).firstElementChild).toBeNull();

  expect(nock.isDone());
});
