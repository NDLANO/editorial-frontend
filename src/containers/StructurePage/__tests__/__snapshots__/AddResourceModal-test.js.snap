// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can paste a valid url and add it to topic 1`] = `
<div
  class="c-taxonomy-lightbox"
>
  <div
    aria-hidden="true"
    class="c-overlay"
  />
  <div
    class="c-taxonomy-lightbox__wrapper"
  >
    <div
      class="c-taxonomy-lightbox__header"
    >
      Søk etter ressurs
      <button
        class="c-button c-button--stripped c-taxonomy-lightbox__close"
      >
        <svg
          fill="currentColor"
          preserveAspectRatio="xMidYMid meet"
          height="1em"
          width="1em"
          class="c-icon"
          viewBox="0 0 24 24"
          data-license="Apache License 2.0"
          data-source="Material Design"
          style="vertical-align: middle;"
        >
          <title>
            Cross
          </title>
          <g>
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="c-taxonomy-lightbox__content"
    >
      <input
        type="text"
        data-testid="addResourceUrlInput"
        placeholder="Lim inn lenke fra ndla.no"
        value="https://beta.ndla.no/subjects/subject:3/topic:1:179373/topic:1:170165/resource:1:168388/"
      />
      <span
        class="c-errorMessage"
      >
        Obs! Denne ressursen er en annen type urn:resourcetype:tasksAndActivities
      </span>
      <div
        data-testid="articlePreview"
        class="c-article-preview"
      >
        <div
          class="c-article-preview__image"
        >
          <img
            src="/placeholder.png"
            alt=""
          />
        </div>
        <div
          class="c-article-preview__content"
        >
          <a
            href="/subject-matter/learning-resource/356/edit/undefined"
          >
            <h1
              class="c-article-preview__title"
            >
              Hva kan du om geologiske prosesser?
            </h1>
          </a>
          <p
            class="c-article-preview__description"
          />
        </div>
      </div>
      <button
        class="c-button c-button--stripped c-taxonomy-lightbox__select-button"
        data-testid="taxonomyLightboxButton"
      >
        Velg
      </button>
    </div>
  </div>
</div>
`;

exports[`Can select a resource from the list and it adds it to topic 1`] = `
<div
  class="c-taxonomy-lightbox"
>
  <div
    aria-hidden="true"
    class="c-overlay"
  />
  <div
    class="c-taxonomy-lightbox__wrapper"
  >
    <div
      class="c-taxonomy-lightbox__header"
    >
      Søk etter ressurs
      <button
        class="c-button c-button--stripped c-taxonomy-lightbox__close"
      >
        <svg
          fill="currentColor"
          preserveAspectRatio="xMidYMid meet"
          height="1em"
          width="1em"
          class="c-icon"
          viewBox="0 0 24 24"
          data-license="Apache License 2.0"
          data-source="Material Design"
          style="vertical-align: middle;"
        >
          <title>
            Cross
          </title>
          <g>
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="c-taxonomy-lightbox__content"
    >
      <input
        type="text"
        data-testid="addResourceUrlInput"
        placeholder="Lim inn lenke fra ndla.no"
        value=""
      />
      <span
        class="c-taxonomy-lightbox__or"
      >
        Eller
      </span>
      <div
        class="c-dropdown"
      >
        <input
          role="combobox"
          aria-autocomplete="list"
          aria-expanded="true"
          autocomplete="off"
          placeholder="Søk på tittel"
          id="downshift-0-input"
          class="c-dropdown__single"
          value=""
        />
        <div
          class="c-dropdown__items c-dropdown__items--resourceMenu"
        >
          <div
            class="c-dropdown__item"
            id="downshift-0-item-0"
          >
            Hva kan du om geologiske prosesser?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-1"
          >
            Hva kan du om petroleumsgeologi?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-2"
          >
            Hva kan du om olje og gass på den norske sokkelen?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-3"
          >
            Hva kan du om letemetoder og datainnsamling?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-4"
          >
            Hva kan du om installasjoner for boring og vedlikehold?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-5"
          >
            Hva kan du om installasjoner for produksjon?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-6"
          >
            Sammendrag om rørhåndtering
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-7"
          >
            Hva kan du om slamsystemet?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-8"
          >
            Hva kan du om sementeringsutstyret?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-9"
          >
            Hva kan du om stigerørsystemet?
          </div>
        </div>
        <button
          class="c-button c-button--stripped c-dropdown__action"
          type="button"
          role="button"
          aria-label="close menu"
          aria-expanded="true"
          aria-haspopup="true"
          data-toggle="true"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon c-icon--medium"
            viewBox="0 0 48 48"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <title>
              Search
            </title>
            <g>
              <path
                d="M31 28h-1.59l-.55-.55C30.82 25.18 32 22.23 32 19c0-7.18-5.82-13-13-13S6 11.82 6 19s5.82 13 13 13c3.23 0 6.18-1.18 8.45-3.13l.55.55V31l10 9.98L40.98 38 31 28zm-12 0c-4.97 0-9-4.03-9-9s4.03-9 9-9 9 4.03 9 9-4.03 9-9 9z"
              />
            </g>
          </svg>
        </button>
      </div>
      <button
        class="c-button c-button--stripped c-taxonomy-lightbox__select-button"
        data-testid="taxonomyLightboxButton"
      >
        Velg
      </button>
    </div>
  </div>
</div>
`;

exports[`Can select a resource from the list and it adds it to topic 2`] = `
<div
  class="c-taxonomy-lightbox"
>
  <div
    aria-hidden="true"
    class="c-overlay"
  />
  <div
    class="c-taxonomy-lightbox__wrapper"
  >
    <div
      class="c-taxonomy-lightbox__header"
    >
      Søk etter ressurs
      <button
        class="c-button c-button--stripped c-taxonomy-lightbox__close"
      >
        <svg
          fill="currentColor"
          preserveAspectRatio="xMidYMid meet"
          height="1em"
          width="1em"
          class="c-icon"
          viewBox="0 0 24 24"
          data-license="Apache License 2.0"
          data-source="Material Design"
          style="vertical-align: middle;"
        >
          <title>
            Cross
          </title>
          <g>
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </g>
        </svg>
      </button>
    </div>
    <div
      class="c-taxonomy-lightbox__content"
    >
      <input
        type="text"
        data-testid="addResourceUrlInput"
        placeholder="Lim inn lenke fra ndla.no"
        value=""
      />
      <span
        class="c-taxonomy-lightbox__or"
      >
        Eller
      </span>
      <div
        class="c-dropdown"
      >
        <input
          role="combobox"
          aria-autocomplete="list"
          aria-expanded="true"
          autocomplete="off"
          placeholder="Søk på tittel"
          id="downshift-0-input"
          class="c-dropdown__single"
          value="Hva kan du om geologiske prosesser?"
        />
        <div
          class="c-dropdown__items c-dropdown__items--resourceMenu"
        >
          <div
            class="c-dropdown__item"
            id="downshift-0-item-0"
          >
            <strong>
              Hva kan du om geologiske prosesser?
            </strong>
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-1"
          >
            Hva kan du om petroleumsgeologi?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-2"
          >
            Hva kan du om olje og gass på den norske sokkelen?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-3"
          >
            Hva kan du om letemetoder og datainnsamling?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-4"
          >
            Hva kan du om installasjoner for boring og vedlikehold?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-5"
          >
            Hva kan du om installasjoner for produksjon?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-6"
          >
            Sammendrag om rørhåndtering
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-7"
          >
            Hva kan du om slamsystemet?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-8"
          >
            Hva kan du om sementeringsutstyret?
          </div>
          <div
            class="c-dropdown__item"
            id="downshift-0-item-9"
          >
            Hva kan du om stigerørsystemet?
          </div>
        </div>
        <button
          class="c-button c-button--stripped c-dropdown__action"
          type="button"
          role="button"
          aria-label="close menu"
          aria-expanded="true"
          aria-haspopup="true"
          data-toggle="true"
        >
          <svg
            fill="currentColor"
            preserveAspectRatio="xMidYMid meet"
            height="1em"
            width="1em"
            class="c-icon c-icon--medium"
            viewBox="0 0 24 24"
            data-license="Apache License 2.0"
            data-source="Material Design"
            style="vertical-align: middle;"
          >
            <title>
              Cross
            </title>
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </g>
          </svg>
        </button>
      </div>
      <div
        data-testid="articlePreview"
        class="c-article-preview"
      >
        <div
          class="c-article-preview__image"
        >
          <img
            src="/placeholder.png"
            alt=""
          />
        </div>
        <div
          class="c-article-preview__content"
        >
          <a
            href="/subject-matter/learning-resource/356/edit/undefined"
          >
            <h1
              class="c-article-preview__title"
            >
              Hva kan du om geologiske prosesser?
            </h1>
          </a>
          <p
            class="c-article-preview__description"
          />
        </div>
      </div>
      <button
        class="c-button c-button--stripped c-taxonomy-lightbox__select-button"
        data-testid="taxonomyLightboxButton"
      >
        Velg
      </button>
    </div>
  </div>
</div>
`;
